Automattic/WP-Job-Manager
# Add dropdown to select author on job listing editor

Fixes #846

~~Based on `change/move-to-select2`. I'll rebase when #1645 is merged.~~ 
![dec-13-2018 17-38-48](https://user-images.githubusercontent.com/68693/49956822-10b13f80-fefe-11e8-89fc-42c1e75220d9.gif)

#### Changes proposed in this Pull Request:

* Add dropdown that is sourced from an AJAX call to select an author on a job listing page.
* Based on Woo's approach with some modifications. I added pagination at the expense of searching `first_name` and `last_name` meta fields.
* Bumped up the priority for the author field so it appears first in the list to fix an issue with `select2` and the block editor. Having it at the bottom of the screen caused an issue with the search input jumping around on activation.
* I tried to make it so we can easily implement this on resumes and applications.
* Fixed a tiny bug that threw a PHP notice when saving a listing that got bumped to a guest post.

#### Testing instructions:

* In both the block and classic editor, change the author of a job listing. Try different searches. 
* Try clearing (`x` on right of input) and saving. 
* Bonus points: go to edit a job listing and then in another tab log out and log in as a low-privileged user (subscriber). Go back to the job listing editor tab (without refreshing) and attempt to change user/search for user. Should fail.


#### Proposed changelog entry for your changes:
* Enhancement: Change job listing authors more easily with dropdown in job listing editor.
